compiler/src/main/scala/org/scalajs/core/compiler/GenJSExports.scala

```diff
@@ -227,6 +227,12 @@ trait GenJSExports extends SubComponent { self: GenJSCode =>
       }
     }
 
+    def genJSConstructorExport(alts: List[Symbol]): js.MethodDef = {
+      withNewLocalNameScope {
+        genExportMethod(alts.map(ExportedSymbol), "constructor")
+      }
```

Comment:
I think this method should *not* use `withNewLocalNameScope`. Its contents will be merged/inlined with the implementations of the init constructors. So they must share the same local name scope. This deserves a comment, also noting that this method is called from a place in `GenJSCode` where it is within a local name scope.